### PR TITLE
NSF funding pilot phase 2

### PR DESCRIPTION
**Pull request by daisieh**
Addresses https://trello.com/c/ebGo5uBp/478-capture-funding-data-at-submission-phase-2.

### CODE REVIEW

**Comment by daisieh** [comment]
In order to set this up on a VM, a bunch of stuff needs to be done. Update your dryad-repo and then execute the following commands:

```
update_schemas.sh
/opt/dryad/bin/dsrun org.dspace.administer.MetadataImporter -f /opt/dryad/config/registries/funder-types.xml
/opt/dryad/bin/dsrun org.dspace.administer.MetadataImporter -f /opt/dryad/config/registries/organization-types.xml
python dspace/etc/upgrades/organization-updates.py
/opt/dryad/bin/dsrun org.datadryad.authority.ImportFunder2Authority
```

The python script updates the database's schemas to reflect the new metadata fields and schemas that were installed in this PR.
The class ImportFunder2Authority imports all of the funder concepts from the FundRef database. If the command gets stuck at "starting", then kill it and restart it: sometimes it can't access the URL. When it works, it will add ~13,300 funder concepts to the database.

**Comment by daisieh** [comment]
Once you've set up the VM, you should go check the NSF concept (funder concept, not the journal concept) and add a fake customer ID number for it. You'll have to go to Manage Scheme>Funder>Search and Add Concept to find it.

Then you can test.

• Create a submission without sponsorship by journal but with a valid NSF grant number.
• Curate it all the way through into the archive. Verify that everything still behaves as expected.
• Go to the item's page. Note the DOI. You should see the updated package page with the new information at the top and the funding info at the bottom, with the metadata.
• Verify that the Datacite metadata for the item contains the funding info:

```
xsltproc dryad-repo/dspace/config/crosswalks/DIM2DATACITE.xsl http://localhost:9999/resource/doi:<your_DOI>/mets.xml > test_datacite.xml
```

• Validate the metadata against the schema:

```
xmllint --noout --schema http://schema.datacite.org/meta/kernel-2.2/metadata.xsd test_datacite.xml
```

Probably it would be good to also test a sponsored-journal submission and a non-sponsored submission and make sure those still act the appropriate way.

**Comment by daisieh** [comment]
Oh! Don't forget that you need to set the new NSF funder concept to have a `organization.paymentPlanType` of `DEFERRED`.

**Comment by ryscher** [comment]
Tested on VM.

Concepts are created correctly. Credit is properly applied for packages associated with NSF grants.

**Comment by ryscher** [comment]
On the item page, package metadata is not displayed. The metadata for files and the citation are correct, but there is an issue generating the display of package's metadata.
screenshot 2016-11-17 00 43 57

**Comment by ryscher** [comment]
On the "show full metadata" screen, the funder information is displayed correctly, but is tagged with an undesirable warning icon.
screenshot 2016-11-17 00 40 59

File: dspace/modules/payment-system/payment-api/src/main/java/org/dspace/workflow/actions/processingaction/FinalPaymentAction.java [hunk 1]
```diff
@@ -79,39 +80,35 @@ public ActionResult execute(Context c, WorkflowItem wfi, Step step, HttpServletR
...
-                    paymentSystemService.sendPaymentApprovedEmail(c,wfi,shoppingCart);
-                    return new ActionResult(ActionResult.TYPE.TYPE_OUTCOME, ActionResult.OUTCOME_COMPLETE);
-                } catch (Exception e) {
-                    log.error(e.getMessage(),e);
-                    paymentSystemService.sendPaymentErrorEmail(c, wfi, shoppingCart, "problem: credit not tallied successfully. \\n \\n " + e.getMessage());
-                    return new ActionResult(ActionResult.TYPE.TYPE_OUTCOME, 2);
+            DryadOrganizationConcept organizationConcept = shoppingCart.getSponsoringOrganization(c);
+            // if journal-based subscription is in place, transaction is paid
+            if (organizationConcept != null) {
+                if (shoppingCart.getJournalSub()) {
```
Comment (ryscher) [inline dspace/modules/payment-system/payment-api/src/main/java/org/dspace/workflow/actions/processingaction/FinalPaymentAction.java]: This line is a little hard to read. Can we change the name of getJournalSub() to hasSubscription() ?

**Comment by ryscher** [comment]
Everything else looks good.

**Comment by daisieh** [comment]
AUGH one problem that I am noting is that this PR is now woefully behind, Spring-wise, so I need to merge those changes in and then we can re-test.

**Comment by daisieh** [comment]
Some problems with your testing: 
1) when you were looking at the package page, were you looking at the "internal-item" view or the "resource/doi" view? The funding information doesn't show up on the internal-item view, but it does show on the public view, so I think that should be sufficient. See first image below.

2) You need to use a valid grant number in order to get a non-scary icon. See second image below.

screen shot 2016-11-18 at 4 46 13 pm

screen shot 2016-11-18 at 4 48 35 pm

File: dspace/modules/payment-system/payment-api/src/main/java/org/dspace/workflow/actions/processingaction/FinalPaymentAction.java (hunk 1 above)
Comment (daisieh) [reply dspace/modules/payment-system/payment-api/src/main/java/org/dspace/workflow/actions/processingaction/FinalPaymentAction.java]: Changed now.

**Comment by daisieh** [comment]
I've added the funding information to the internal item view as well.

**Comment by ryscher** [comment]
I think my issues might be arising from the way I resolved the merge conflicts. For my current merge attempt, see #1250 -- this merge doesn't build, but I haven't looked into the issue yet.

**Comment by ryscher** [comment]
Tested updates on VM. Everything looks good. This is ready to merge any time.

**Pull request merged**